### PR TITLE
chore(http-handler): Track execution + certification time for all ingress messages

### PR DESCRIPTION
**Pull request by DSharifi**
We use the `IngressWatcher` in call v2 to record the durations for a message to complete execution and be added to the certified state. A tokio task is spawned for each request, where the spawned task waits in the background for the message to be certified.

### CODE REVIEW

**Review by mraszyk — APPROVED** [review]

**Comment by tthebst** [comment]
Is there a reason why we want to measure this?

**Comment by DSharifi** [comment]
> Is there a reason why we want to measure this? Is it to do a comparison to v3?

It will give us a proxy for the E2E latency of all ingress messages. At least from the view of the local replica.

**Comment by tthebst** [comment]
Isn't it the same as the v3 endpoint latency?

**Comment by DSharifi** [comment]
> Isn't it the same as the v3 endpoint latency?

Yes. But we don't have any traffic there yet. We want to collect metrics already now with the v2 endpoint so we have a baseline to compare to when we roll out future improvements, such as the reduced notarisation delays.

**Review by tthebst — APPROVED** [review]

File: rs/http_endpoints/public/src/call/call_v2.rs [hunk 1]
```diff
@@ -17,48 +17,107 @@ use ic_types::{
...
+    // We spawn a task to register the certification time of the message.
+    // The subscriber in the spawned task records the certification time of the message
+    // when `wait_for_certification` is called.
+    if let Some(ingress_watcher_handle) = ingress_watcher_handle {
+        tokio::spawn(async move {
+            // We acquire a permit to bound the number of concurrent tasks. If no permits are available,
+            // we return early to terminate the task.
+            let ingress_tracking_permit = ingress_tracking_semaphore.try_acquire();
+            let Ok(_permit) = ingress_tracking_permit else {
+                return;
```
Comment (rumenov) [inline rs/http_endpoints/public/src/call/call_v2.rs:103]: nit: I would print a log warning or have a metric when this happens

**Review by rumenov — APPROVED** [review]

**Pull request merged**